File: backend/src/main/java/com/back/back9/domain/websocket/vo/CandleInterval.java
package com.back.back9.domain.websocket.vo;

import lombok.Getter;

public enum CandleInterval {
    SEC("seconds", 1000),
    MIN_1("minutes/1", 1000),
    MIN_30("minutes/30", 1000),
    HOUR_1("minutes/60", 1000),
    DAY("days", 500),
    WEEK("weeks", 150),
    MONTH("months", 50),
    YEAR("years", 10);

    private final String redisKeySuffix;
    @Getter
    private final int maxSize;

    CandleInterval(String redisKeySuffix, int maxSize) {
        this.redisKeySuffix = redisKeySuffix;
        this.maxSize = maxSize;
    }

    public String redisKey(String symbol) {
        return symbol + ":" + redisKeySuffix;
    }

    public String getSuffix() {
        return redisKeySuffix;
    }

    @Override
    public String toString() {
        return redisKeySuffix;
    }

    public static CandleInterval fromWebSocketType(String websocketType) {
        return switch (websocketType) {
            case "candle.1s" -> SEC;
            case "candle.1m" -> MIN_1;
            case "candle.30m" -> MIN_30;
            case "candle.1h" -> HOUR_1;
            case "candle.1d" -> DAY;
            case "candle.1w" -> WEEK;
            case "candle.1M" -> MONTH;
            case "candle.1y" -> YEAR;
            default -> throw new IllegalArgumentException("Unknown WebSocket candle type: " + websocketType);
        };
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/websocket/vo/CandleInterval.kt
package com.back.back9.domain.websocket.vo

/**
 * Upbit API의 캔들 종류와 Redis 키 생성을 관리하는 Enum 클래스
 */
enum class CandleInterval(val suffix: String, val webSocketType: String) {
    SEC("minutes/1", "candle.1s"),
    MIN_1("minutes/1", "candle.1m"),
    MIN_3("minutes/3", ""),
    MIN_5("minutes/5", ""),
    MIN_10("minutes/10", ""),
    MIN_15("minutes/15", ""),
    MIN_30("minutes/30", ""),
    HOUR_1("minutes/60", ""),
    HOUR_4("minutes/240", ""),
    DAY("days", ""),
    WEEK("weeks", ""),
    MONTH("months", "");

    /**
     * Redis 저장을 위한 고유 키를 생성합니다. (예: "CANDLE:MIN_1:KRW-BTC")
     */
    fun redisKey(marketCode: String): String {
        return "CANDLE:${this.name}:$marketCode"
    }

    companion object {
        fun fromWebSocketType(type: String): CandleInterval {
            return entries.find { it.webSocketType == type }
                ?: throw IllegalArgumentException("Unsupported WebSocket candle type: $type")
        }
    }
}